src/components/Header/index.tsx

```diff
@@ -38,11 +33,24 @@ const Header = () => {
     const signIn = async () => {
         await authService.signIn();
         session.setAuthenticated(true);
+        setIsAuthenticated(true);
+    };
+
+    const signOut = async () => {
+        await authService.signOut();
+        session.setAuthenticated(false);
+        setIsAuthenticated(false);
```

Comment:
not directly related to this PR, but I'm curious if there is a specific reason we need both the global state (via `useSession()`) and the local `useState` in this component. Is there a case where the local state provides a benefit that the global state does not? If not, would it make sense to refactor to use only `useSession()` for better consistency and simplicity?

Reply:
I won't be able to answer right now why but the state is using authService value, not the session one. So I would leave it as is for now and later think if we can improve it.